Layerscape: update kernel patches and all packages to LSDK-18.06

yangbolu1991 opened the pull request:
NXP had announced its OpenWrt-18.08 release. Please see announcement and release note attached.
This pull request cherry-picked and rebased these released patches. Please help to review and merge.
Thanks a lot.

NXP LSDK homepage.
https://www.nxp.com/products/processors-and-microcontrollers/applications-processors/qoriq-platforms/t-series/layerscape-software-development-kit:LAYERSCAPE-SDK

[Announcement.docx](https://github.com/openwrt/openwrt/files/2292410/Announcement.docx)
[Release_notes.txt](https://github.com/openwrt/openwrt/files/2292411/Release_notes.txt)

yangbolu1991 committed:
layerscape: update fman-ucode to LSDK-18.06

Actually there was no change for fman-ucode in LSDK-18.06
just tagged with LSDK-18.06. This patch is to rework the
fman-ucode makefile to make it more readable, and to use
lsdk-1806 as the PKG_VERSION.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update ls-dpl to LSDK-18.06

The dpl-examples source code had been migrated to
codeaurora for LSDK-18.06 release and the future
release. This patch is to update this package to
LSDK-18.06.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update ls-mc to LSDK-18.06

This patch is to update ls-mc to LSDK-18.06 release.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update ppfe-firmware to LSDK-18.06

This patch is to update ppfe-firmware to LSDK-18.06 release.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update ls-rcw to LSDK-18.06

The rcw source code had been migrated to codeaurora
for LSDK-18.06 release and the future release. The
source code had also involved ls1012ardb/ls1012afrdm/
ls1088ardb/ls2088ardb rcw, so we updated ls-rcw to
LSDK-18.06, reworked the makefile and dropped ls-rcw-bin
package in this patch. Also reworked ls-rcw patch to
adapt to the latest source code.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update restool to LSDK-18.06

The restool source code had been migrated to codeaurora
for LSDK-18.06 release and the future release. This patch
is to update restool to LSDK-18.06 release.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update ls-ppa to LSDK-18.06

This patch is to update ls-ppa to LSDK-18.06 release
and to rework ls-ppa makefile to make it more readable.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update u-boot to LSDK-18.06

The u-boot source code had been migrated to codeaurora
for LSDK-18.06 release and the future release. This
patch is to update u-boot to LSDK-18.06 for both
uboot-layerscape and uboot-layerscape-armv8_32b packages.
Besides, this patch also introduced some other changes.
- Reworked uboot-layerscape makefile to make it more
  readable.
- Define package in uboot-layerscape-armv8_32b for each board.
- Fixed u-boot package selection in target image makefile.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: define ls-append function

This patch is to define a ls-append function for
each device to reuse it for image appending.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: add armv7 subtarget and ls1021atwr board support

The NXP TWR-LS1021A module is a development system based
on the QorIQ LS1021A processor.
- This feature-rich, high-performance processor module can
  be used standalone or as part of an assembled Tower System
  development platform.
- Incorporating dual Arm Cortex-A7 cores running up to 1 GHz,
  the TWR-LS1021A delivers an outstanding level of performance.
- The TWR-LS1021A offers HDMI, SATA3 and USB3 connectors as
  well as a complete Linux software developer's package.
- The module provides a comprehensive level of security that
  includes support for secure boot, Trust Architecture and
  tamper detection in both standby and active power modes,
  safeguarding the device from manufacture to deployment.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: drop uboot-layerscape patches

Dropped uboot-layerscape patches which were environemnt patches.
We will make u-boot environment binaries with a txt file for all
devices.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: add u-boot environment support for OpenWrt boot

This patch is to implement u-boot environment txt files
to support OpenWrt boot for all layerscape devices.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: add SD card boot support

NOR/QSPI Flash on Layerscape board only has limited 64MB memory size.
Since some boards (ls1043ardb/ls1046ardb/ls1088ardb/ls1021atwr)
could support SD card boot, we added SD boot support for them to put
all things on SD card to meet large memory requirement.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: add ls1012afrwy support and drop ls1012afrdm

ls1012afrdm was no longer supported in NXP Layerscape SDK.
Instead a new board ls1012afrwy was introduced in LSDK.
This patch is to drop ls1012afrdm and add ls1012afrwy support.
Since only 2MB NOR flash could be used, we just put u-boot
and firmware on NOR flash, and put kernel/dtb/rootfs on SD
card.

The Layerscape FRWY-LS1012A board is an ultra-low-cost
development platform for LS1012A Series Communication
Processors built on Arm Cortex-A53. This tool refines the
FRDM-LS1012A with more features for a better hands-on experience
for IoT, edge computing, and various advanced embedded
applications. Features include easy access to processor I/O,
low-power operation, micro SD card storage, an M2 connector, a
small form factor, and expansion board options via mikroBUS Click
Module. The MicroBUS Module provides easy expansion via hundreds
of powerful modules supporting sensors, actuators, memories,
and displays.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update README for SD card boot and new devices

This patch is to update the README since SD card boot support
and LS1012AFRWY/LS1021ATWR were introduced.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: fix get_device_file() function of restool

The restool failed to work with current gcc-7.3.0-musl.
This patch is to add a restool fix-up patch to fix
multiple problems encountered in the get_device_file()
function:
 - The deprecated atoi() function is replaced by strtoul
 - An invalid memory access was being performed by using
 memory from dir->d_name even after closedir(). This is
 fixed by a strdup() on the device filename.
 - Also, error prints now print any relevant error code.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: update linux 4.9 patches to LSDK-18.06

This patch is to update linux 4.9 patches to LSDK-18.06
release and to adjust config-4.9 accordingly.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 committed:
layerscape: split image makefile per subtarget

This patch is to split image makefile per subtarget.
The ARMv7 subtarget will be added in the future.
It will be not convinient if only one makefile is used
for several subtargets management and future development.
This patch also dropped 32-bit Traverse LS1043-S since
Traverse only intended to support 64-bit and the 32-bit
compile now had an issue.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

pepe2k commented:
@yangbolu1991 

> Signed-off-by: Yangbo Lu <yangbo.lu>

This doesn't look like a correct e-mail address.

Cheers,
Piotr

yangbolu1991 commented:
Fixed Signed-off-by.

yangbolu1991 commented:
Any comments?
Thanks.

hauke commented on package/firmware/layerscape/fman-ucode/Makefile:
```diff
@@ -6,84 +6,48 @@
 
-PKG_BUILD_DIR=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_VERSION)
-
-PKG_FLAGS:=nonshared
```
@jow-  will this still work when the noshared is removed?

hauke commented on package/boot/uboot-layerscape/Makefile:
```diff
@@ -6,16 +6,15 @@
-PKG_SOURCE_URL:=https://github.com/qoriq-open-source/u-boot.git
-PKG_SOURCE_VERSION:=9f7df1b406ff11409021cd2112beedd6b57bb600
-PKG_MIRROR_HASH:=b3756f814b731af2d03b0582ece90b2de564955b778f341fbfc34fa9bd849819
+PKG_SOURCE_URL:=https://source.codeaurora.org/external/qoriq/qoriq-components/u-boot
```
How long will your repositories be available at https://source.codeaurora.org ? 
I though the merger between NXP and Qualcomm stopped?

yangbolu1991 replied:
Yes, you're right.
But as I know we will still use CAF for the futhure layerscape release.

hauke commented on target/linux/layerscape/armv7/config-4.9:
```diff
@@ -0,0 +1,1076 @@
+# CONFIG_AXP20X_POWER is not set
+CONFIG_BATTERY_ACT8945A=y
+CONFIG_BATTERY_SBS=y
+CONFIG_BCMA=y
```
Is BCMA really needed, if not please remove it?

yangbolu1991 replied:
Will remove it.

hauke commented on target/linux/layerscape/armv7/config-4.9:
```diff
@@ -0,0 +1,1076 @@
+# CONFIG_NET_VENDOR_TEHUTI is not set
+# CONFIG_NET_VENDOR_TI is not set
+# CONFIG_NET_VENDOR_VIA is not set
+# CONFIG_NET_VENDOR_WIZNET is not set
```
Please do not deactivate all these network vendor directories, the generic configuration should take care of this

yangbolu1991 replied:
Get it. Will fix them.
Just a question, when I make the kernel config for subtarget, is there any better idea to know which configs are taken care by openwrt, than manual checking the  generic config, kernel module package, and config/Config-kernel.in?

hauke replied:
No not really, because there could be a valid reason why you want to overwrite the config.
You should run later: "make kernel_oldconfig CONFIG_TARGET=subtarget" over the config, which will remove all unnecessary config options.

hauke commented on target/linux/layerscape/armv7/config-4.9:
```diff
@@ -0,0 +1,1076 @@
+CONFIG_RWSEM_XCHGADD_ALGORITHM=y
+CONFIG_SATA_AHCI=y
+CONFIG_SATA_AHCI_PLATFORM=y
+CONFIG_SATA_MV=y
```
Do you have a Marvell SATA controller in the SoC or on the board?

yangbolu1991 replied:
Sorry, no. Will remove it.

hauke commented on target/linux/layerscape/armv7/config-4.9:
```diff
@@ -0,0 +1,1076 @@
+CONFIG_IOMMU_SUPPORT=y
+CONFIG_IOSCHED_CFQ=y
+CONFIG_IPC_NS=y
+CONFIG_IPV6=y
```
The generic kernel configuration should take care of these network settings, when you change then here it could cause some problems when users activate some kernel modules . 
Please remove all options which are not needed for your SoC or the board you are running from the configuration file.

yangbolu1991 replied:
Will clean up the kernel config. Thanks a lot.

hauke reviewed (commented):
The next OpenWrt release will ship only with kernel 4.14. What are you plans to update the Linux kernel to version 4.14?

yangbolu1991 commented on package/boot/uboot-layerscape-armv8_32b/Makefile:
```diff
@@ -17,37 +17,70 @@ PKG_RELEASE:=1
+PKG_MIRROR_HASH:=5fcb58c14cdc934793ff315e178ad1d9d2ff55fcaac394b48ec116d1b11ca324
 
-PKG_MAINTAINER:=Yangbo Lu <yangbo.lu@nxp.com>
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
Get it. Will remove this line using the default.

yangbolu1991 commented on package/boot/uboot-layerscape-armv8_32b/Makefile:
```diff
@@ -8,7 +8,7 @@
 
 PKG_NAME:=uboot-layerscape-armv8_32b
-PKG_SOURCE_DATE:=2018-01-22
+PKG_VERSION:=lsdk-1806
```
Yes. Since the packages were from LSDK release,  I'd like to use the release tag as the PKG_VERSION.
When upgrade these packages, we update them to the commit which has the release tag.

yangbolu1991 commented:
@hauke Thanks a lot for your comments.
Regarding to linux 4.14, actually NXP LSDK release already included a linux-4.14 tree.
We can support linux-4.14 in OpenWrt anytime. I could start to work on it.
May I know when will be the next OpenWrt release?

yangbolu1991 commented:
Updated to v2 patch-set.
Changes for v2:
- used default PKG_BUILD_DIR for layerscape packages.
- cleaned up kernel config for new armv7 subtarget.

yangbolu1991 commented:
Rebased patches.

yangbolu1991 commented:
Any comments on the new version patch-set?
Thanks.

mcbridematt commented on package/firmware/layerscape/ls-dpl/Makefile:
```diff
@@ -8,40 +8,36 @@
 endef
 
+MAKE_PATH:=config
+
```
ls-dpl fails to build if the device tree compiler ('dtc') is not present on the host system.

I suggest using the kernel copy of 'dtc', by adding:
```MAKE_VARS+= PATH=$(LINUX_DIR)/scripts/dtc:$(PATH)```
as well as:
```include $(INCLUDE_DIR)/kernel.mk```
near the top of the file (to get $(LINUX_DIR))

yangbolu1991 replied:
Thanks a lot, Mathew.
I just added a fix-up patch for this.

yangbolu1991 committed:
layerscape: build ls-dpl package with linux dtc tool

Building ls-dpl package requires the dtc tool. This patch
is to support using linux dtc tool for ls-dpl package.
This avoids compile issue when host system doesn't have
the dtc tool.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 commented:
+ @blogic 
Hi John, 

could you help to review and merge this pull request?
Thanks a lot.

yangbolu1991 commented:
Any comments? Thanks.

hauke commented:
The build bot found a build problem in the 64 bit images, see here:
http://phase1.builds.lede-project.org/builders/layerscape%2Farmv8_64b/builds/559/steps/images/logs/stdio
```
Error: /var/lib/buildbot/slaves/slashdirt-02/MAIN/build/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.9.128/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb.dts:175.29-30 syntax error
FATAL ERROR: Unable to parse input tree
```

yangbolu1991 commented:
@hauke 
Strange. I couldn't reproduce this issue. Below were my steps.
./scripts/feeds update -a
./scripts/feeds install -a
make menuconfig (select all layerscape armv8_64b targets)
make download -j16
make -j16
make -j12 target/install V=s

Regarding to the error dts build (fsl-ls1046a-rdb.dts), actually the ls1046ardb device used fsl-ls1046a-rdb-sdk.dts, not fsl-ls1046a-rdb.dts.
Any idea to reproduce this issue?

Thanks.

hauke commented:
I only saw this sometimes with parallel builds and also the build bots only see it sometimes it looks like some parallel build issue.
It is only happening sometimes:
http://phase1.builds.lede-project.org/builders/layerscape%2Farmv8_64b
